core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -217,12 +233,16 @@ private[spark] class SparkSubmit extends Logging {
    */
   private[deploy] def prepareSubmitEnvironment(
       args: SparkSubmitArguments,
-      conf: Option[HadoopConfiguration] = None)
+      conf: Option[HadoopConfiguration] = None,
+      sparkConfOpt: Option[SparkConf] = None)
```

Comment:
Do you need this extra argument? Can't you achieve the same by passing `--conf` arguments in the unit test?

Reply:
There're couple of places which configuration is being set, and the place I hit the issue was configuration file, so I was trying to make sure it is same. But yeah it would be effectively testing the same (since `--conf` will be handled before) so no big deal. Will roll back the change and use `--conf`.

Reply:
You can test the config file path if you want to. Just create a file and use `--properties-file`.